test/e2e/far_e2e_test.go

```diff
@@ -116,30 +117,41 @@ var _ = Describe("FAR E2e", func() {
 			nodeBootTimeBefore, err = e2eUtils.GetBootTime(clientSet, nodeName, testNsName, log)
 			Expect(err).ToNot(HaveOccurred(), "failed to get boot time of the node")
 
-			// create tested pod, and save its creation time
-			// it will be deleted by FAR CR
+			// create tested pod which will be deleted by the far CR
 			pod = e2eUtils.GetPod(nodeName, testContainerName)
 			pod.Name = testPodName
 			pod.Namespace = testNsName
 			Expect(k8sClient.Create(context.Background(), pod)).To(Succeed())
-			log.Info("Tested pod has been created", "pod", testPodName)
-			creationTimePod = metav1.Now().Time
 			DeferCleanup(cleanupTestedResources, pod)
 
 			// set the node as "unhealthy" by disabling kubelet
 			makeNodeUnready(selectedNode)
 
+			startTime = time.Now()
 			far := createFAR(nodeName, fenceAgent, testShareParam, testNodeParam)
-			DeferCleanup(deleteFAR, far)
+			DeferCleanup(deleteFAR, startTime, far)
 		})
 		When("running FAR to reboot two nodes", func() {
 			It("should successfully remediate the first node", func() {
-				checkRemediation(nodeName, nodeBootTimeBefore, creationTimePod, pod)
+				checkRemediation(nodeName, nodeBootTimeBefore, pod)
+				remediationTimes = append(remediationTimes, time.Since(startTime))
 			})
 			It("should successfully remediate the second node", func() {
-				checkRemediation(nodeName, nodeBootTimeBefore, creationTimePod, pod)
+				checkRemediation(nodeName, nodeBootTimeBefore, pod)
+				remediationTimes = append(remediationTimes, time.Since(startTime))
 			})
 		})
+		AfterEach(func() {
+			if len(remediationTimes) == 2 {
```

Comment:
I didn't find a better way to check that both tests are over. `AfterAll` didn't work for me (or I didn't use it properly).

Reply:
If I read [1] correctly, `AfterAll` can only be used in `Ordered` containers, which cannot be randomized though, so considering the latest changes, it doesn't seem a viable option.

[1] https://onsi.github.io/ginkgo/#setup-in-ordered-containers-beforeall-and-afterall

Reply:
have a look at `AfterSuite`